src/Message.ts

```diff
@@ -254,3 +242,126 @@ export class MessageV2 extends MessageBase implements proto.MessageV2 {
 }
 
 export type Message = MessageV1 | MessageV2
+
+export interface DecodedMessage {
+  id: string
+  messageVersion: 'v1' | 'v2'
+  senderAddress: string
+  conversation: Conversation
+  contentType: ContentTypeId
+  sent: Date
+  contentTopic: string
```

Comment:
contentTopic seems to be specific to V1, do we want to make it part of the interface?

Reply:
So, I think in the context of V2 messages it is also relevant (especially in a world where we have stitching and the reference to the Conversation is not sufficient to know whether the message came from the latest topic).

It's also strictly necessary to figure out if a message came from the DM or Intro topic in `streamAllMessages`.

Reply:
Good point about stitching, but we could introduce it later when we get to it.

The streamAll aspect is what I meant that it's a V1 only concern. So I still feel it would be better to leave this out for now.

Reply:
OK, fair enough. It's not really that harmful to have it. It was more about trying to keep the interface as tidy as possible.